source/postprocess/particles.cc

```diff
@@ -97,16 +97,16 @@ namespace aspect
               }
 
             // If the property has dim components, we treat it as vector
-            if (n_components == dim)
+            if (n_components == dim && property_index_to_output_index[field_position] > 0)
```

Comment:
I do not see why this should fix the error. If `property_index_to_output_index[field_position] == 0` we either have the first field (in which case we *do* want to add the field name to the list of `vector_datasets`), or we have a field that should be excluded (in which case we never get here, because line 84 is triggered). To me it looks like there is an off-by-one error somewhere else, and you fix it by omitting something else here. Have you checked in the vtu output if the name of a particle property is correct for the data that is written?